src/stan/lang/generator/get_typedef_var_type.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef STAN_LANG_GENERATOR_GET_TYPEDEF_VAR_TYPE_HPP
+#define STAN_LANG_GENERATOR_GET_TYPEDEF_VAR_TYPE_HPP
+
+#include <stan/lang/ast.hpp>
+#include <string>
+
+namespace stan {
+  namespace lang {
+    /**
+     * Return cpp type name or typedef used for bare_expr_type.
+     * NOTE: (26/02/2018)
+     *   typdefs come from `lib/stan_math/prim/mat/fun/typedefs.hpp`
+     *   vectors and matrices of double (appropriate for data, not params)
+     *
+     * @param[in] bare_type bare_type
+     */
+    std::string
+    get_typedef_var_type(const bare_expr_type bare_type) {
```

Comment:
This argument should be a constant reference---there's an `&` missing.  I'm also not sure why this function makes a copy of the argument---is it because those methods like `is_array_type()` are not declared `const`?  If so, the fix is to make those `const` methods, not copy to a mutable bare expression type here.